### ISSUE TITLE
Iteration time increases dramatically. 42s/it

### ISSUE THREAD

**Issue by logogriph2** [huzeyann/BrainDecodesDeepNets#1]
Hello!
When I'm running an example, time for the fist iteration is around 5s. For 15 step it is already 20-30s 
`Epoch 0:   1%|▏         | 16/1231 [11:04<14:01:28, 41.55s/it, loss=0.412, v_num=10]`

Changing num_workers doesn't help. Setting flag on_epoch=False in training_step logging also doesn't help.

When I wrote a custom training loop without lightning, iteration time is about 3-9s:
```
15 iter: 3.1s
....
90 iter: 6.6s
```
Same problem with a [MEM](https://github.com/huzeyann/MemoryEncodingModel)

env: ubuntu 22.04, python 3.9.18, torch-2.0.0, cuda-11.7, lightning 1.9.4
hardware: GTX1080, 8Gb VRAM, 48Gb RAM, code and data are on the same SSD

**Comment by huzeyann** [comment]
This might have something to do with pytorch-lightning training loop, if you custom training loop is implemented without pytorch-ligntning.

I don't have a good idea on the exact cause of the issue, are you running `example.ipynb` or `plmodel.py`?

In my code, num_workers and trainer flags are already set to good values.

The only thing come to my mind is you are using 1080, which does not support FP16. My code in `example.ipynb` use FP16 by default, this might cause issue, you can change to FP32 by editing
```python
trainer = pl.Trainer(
    max_epochs=10,
    accelerator="gpu",
    devices=[0],
    gradient_clip_val=0.5,
    precision=16,  # change 16 to 32
    limit_train_batches=1.0,
    limit_val_batches=1.0,
    enable_checkpointing=False,
)
```

**Comment by huzeyann** [comment]
The [MEM](https://github.com/huzeyann/MemoryEncodingModel) repository was not designed to fit into 8GB memory, you need 24GB at least; this repository is designed to fit 8GB VRAM

**Comment by huzeyann** [comment]
This might have something to do with caching, you could try turn off caching by
```python
plmodel = PLModel(cfg, backbone, cached=False)
```

by default, cached features is stored in cpu after each forward call

**Comment by logogriph2** [comment]
Thank you! That helped!

**Comment by logogriph2** [comment]
Fun fact: with `cached=False` it runs even in default colab

**Comment by logogriph2** [comment]
> The [MEM](https://github.com/huzeyann/MemoryEncodingModel) repository was not designed to fit into 8GB memory, you need 24GB at least; this repository is designed to fit 8GB VRAM

with batch_size=1 and a gradient accumulation MEM runs with 8Gb RAM

**Comment by huzeyann** [comment]
you need to increase `REGULARIZATION.DECAY_TOTAL_STEPS` by 8x if you decrease batch size 8x and use 8 gradient accumulation, otherwise the model will converge into local minimal (checkout my paper table 15 and figure 24). In general a larger decay total steps is better.
